### ISSUE TITLE
Exported attributes for vcd_vapp resource (vDirector)

### ISSUE THREAD

**Issue by felnne** [hashicorp/terraform#4935]
Hi,

I'm not sure if this has already been discussed and i'm just not seeing something obvious, but is it possible to get any details (i.e. outputs) for vApp resources in vDirector?

My use-case would be to determine the IPv4 address of the first NIC of the first VM within a vApp, such that I can then create a DNS record (as you would with say an EC2 EIP in AWS).

In my case i'd be using Route53 for creating the DNS record but I don't imagine that matters. I am relatively new to vDirector, so i'm not sure if I am missing a more typical way this would be done that means I don't need to know the IP address of each VM.

If it were possible to expose the same outputs as a [vSphere](https://www.terraform.io/docs/providers/vsphere/r/virtual_machine.html) VM, which I don't use but look ideal from the documentation, that would be great. Looking at the *govcloudair* documentation, which I understand is fairly similar to the vdirector library used for this provider, it looks like there is a resource that would be able to return this information.

However since a vApp could contain multiple VMs, each containing network interfaces i'm not sure if this is something that would be considered too complex or unknowable.

This example gives a more specific example of how I would intend to use a feature like this if it existed:

```tf
# Example VM - Only accessible privately
#
# This resource implicitly depends on outputs from the the 'terraform_remote_state.BAS-JASMIN' resource
#
# Terraform source: https://www.terraform.io/docs/providers/vcd/r/vapp.html
resource "vcd_vapp" "example-dev-node1" {
    name          = "example-dev-node1"
    catalog_name  = "${terraform_remote_state.BAS-JASMIN.output.BAS-JASMIN-vApp-Template-Catalogue-Name}"
    template_name = "antarctica/trusty-vdirector-0.1.2"
    memory        = 512
    cpus          = 1

    network_name  = "${terraform_remote_state.BAS-JASMIN.output.BAS-JASMIN-Organisation-Network-Name}"

    metadata {
        Name         = "example-dev-node1"
        X-Managed-By = "Terraform"
        X-Project    = "scratch-test"
        X-Purpose    = "node"
    }
}

# This resource implicitly depends on the 'vcd_vapp.example-dev-node1' resource
# This resource implicitly depends on outputs from the the 'terraform_remote_state.BAS-JASMIN' resource
#
# AWS source: http://docs.aws.amazon.com/Route53/latest/DeveloperGuide/rrsets-working-with.html
# Terraform source: https://www.terraform.io/docs/providers/aws/r/route53_record.html
#
# Tags are not supported by this resource
resource "aws_route53_record" "example-dev-node1-int" {
    zone_id = "${terraform_remote_state.BAS-JASMIN.output.BAS-JASMIN-Internal-Subdomain-ID}"

    name = "example-dev-node1"
    type = "A"
    ttl = "300"

    records = [
        "${vcd_vapp.example-dev-node1.children.0.NetworkConnectionSection.0.IPAddress}"
    ]
}
```

The syntax for referencing the IP address of the first network connection within the first VM of a vApp, is purely illustrative.

I'm not sure if this the right place to ask for feature requests - so if there's somewhere else I should redirect this question to please let me know. I had a look for older vDirector issues on here and on the Google Group but couldn't see anything relevant.

Thanks,
Felix.

**Comment by robcoward** [comment]
Hi @felnne, I've just raised #5159 which may also provide you with a solution. If you provide a static ip when defining your vapp, you can reference it as vcd_vapp.example-dev-node1.ip but when left to DHCP, the ip address allocated to the vm is not known to vcloud director until the open-vm-tools daemon run on the booted vm, after which a terraform refresh does retrieve the respective ip address for you (although too late to be used within the same terraform apply session).

**Comment by felnne** [comment]
Hi @robcoward, thanks for that, from that example it looks to solve my issue completely!

Think I will leave this open until I have a chance to test when this is available.

**Comment by robcoward** [comment]
Hi @felnne. #5195 has been released in v0.6.12 - has this solved your problem for you ?

**Comment by felnne** [comment]
Hi, sorry for the delay - yes that now works.

Thanks!
